dbt_subprojects/daily_spellbook/models/_metrics/fees/chains/solana/metrics_solana_gas_fees_daily_address.sql

```diff
@@ -0,0 +1,78 @@
+{% set blockchain = 'solana' %}
+
+{{ config(
+        schema = 'metrics_' + blockchain
+        , alias = 'gas_fees_daily_address'
+        , materialized = 'incremental'
+        , file_format = 'delta'
+        , incremental_strategy = 'merge'
+        , unique_key = ['blockchain', 'block_date', 'address']
+        , incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.block_date')]
+        )
+}}
+
+with fees as (
+    select
+        blockchain
+        , block_date
+        , tx_from as address
+        , sum(tx_fee_usd) as gas_fees_usd
+    from
+        {{ source('gas', 'fees') }}
+    where
+        blockchain = '{{ blockchain }}'
+        {% if is_incremental() %}
+        and {{ incremental_predicate('block_date') }}
+        {% endif %}
+    group by
+        blockchain
+        , block_date
+        , tx_from
+), solana_vote_fees as (
+    -- solana vote fees are stored in a different spell due to data volume & lack of value-add for materializing the fee breakdown
+    select
+        blockchain
+        , block_date
+        , from_base58(signer) as address
+        , sum(tx_fee_usd) as gas_fees_usd
+    from
+        {{ source('gas_solana', 'vote_fees') }}
+    where
+        1 = 1
+        {% if is_incremental() %}
+        and {{ incremental_predicate('block_date') }}
+        {% endif %}
+    group by
+        blockchain
+        , block_date
+        , from_base58(signer)
+), combined_fees as (
+    select
+        coalesce(fees.blockchain, solana_vote_fees.blockchain) as blockchain
+        , coalesce(fees.block_date, solana_vote_fees.block_date) as block_date
+        , coalesce(fees.address, solana_vote_fees.address) as address
+        , coalesce(fees.gas_fees_usd, 0) + coalesce(solana_vote_fees.gas_fees_usd, 0) as gas_fees_usd
+    from
+        fees
+    full outer join
+        solana_vote_fees
+        on fees.blockchain = solana_vote_fees.blockchain
+        and fees.block_date = solana_vote_fees.block_date
+        and fees.address = solana_vote_fees.address
+)
+select
+    fees.blockchain
+    , fees.block_date
+    , fees.address
+    , coalesce(od.name, 'Unknown') as name
+    , coalesce(od.primary_category, 'Uncategorized') as primary_category
+    , coalesce(od.country_name, 'Unknown') as hq_country
+    , fees.gas_fees_usd
+from combined_fees as fees
+left join
+    {{ source('labels', 'owner_addresses') }} as oa
+    on fees.blockchain = oa.blockchain
+    and fees.address = oa.address
+left join
+    {{ source('labels', 'owner_details') }} as od
+    on oa.owner_key = od.owner_key
```

Comment:
I've seen this join structure everywhere where labels is used, wonder if we should put it in a model on top?

Reply:
do you mean put logic into a macro and reuse, to have logic in one spot?
or rather, don't materialize labels data and have view at end of each which joins to it (performance in mind)?

Reply:
Sorry I should have been more specific, I think we should maybe make a `labels.some_better_name` view model that looks like
```
select 
*
from labels.owner_addresses
left join labels.owner_details using (owner_key)
```
Then you only need to join this on the blockchain and address and you get the right details for any address.
This was more a generic comment when seeing labels used, nothing to be addressed in this PR.

Reply:
noted, good feedback! might be worth applying here, if we're blocked on merge anyway.